astropy/io/fits/fitstime.py

```diff
@@ -259,13 +259,20 @@ def _verify_column_info(column_info, global_info):
     # considered.
     elif global_info['TREFPOS'] == 'TOPOCENTER':
 
-        column_info['location'] = global_info['location']
-        if column_info['location'] is None:
+        if global_info['location'] is None:
             warnings.warn(
                 'Time column reference position "TRPOSn" is not specified. The '
                 'default value for it is "TOPOCENTER", but due to unspecified '
                 'observatory position, reference position will be ignored.',
                 AstropyUserWarning)
+        else:
+            warnings.warn(
+                'Time column reference position "TRPOSn" is not specified. The '
+                'default value for it is "TOPOCENTER", and the observatory position '
+                'has been specified. However, for supporting column-specific location, '
+                'reference position will be ignored for this column.',
+                AstropyUserWarning)
+        column_info['location'] = None
```

Comment:
As above, not clear why this behavior is being changed to *always* ignore global location.

Reply:
The problem is when there are multiple time columns in a table, one with a location associated with it and one without one.

```
t = Table()
t['a'] = Time(['2000:001'], format='isot', scale='utc', location=EarthLocation(-2446354,
                      4237210, 4077985, unit='m'))
t['b'] = Time([1], format='cxcsec', scale='tt')
```

The FITS time standard requires us to write the EarthLocation(-2446354, 4237210, 4077985, unit='m') in global time keywords OBSGEO-X, OBSGEO-Y, OBSGEO-Z. TRPOSn is a column-specific keyword and we would set it for the column 'a' as TRPOS1 = 'TOPOCENTER'. But what FITS would do is that if there is no value for TRPOS2 (for column 'b'), by default it would treat it as TRPOS2 = 'TOPOCENTER' and use the global keywords OBSGEO-X, OBSGEO-Y, OBSGEO-Z as its location. But, we want reproducibility with astropy, so instead when TRPOSn is not specified, I treat the default as ``None`` instead of 'TOPOCENTER'. I think this makes sense, as the user should ideally specify ``TRPOSn`` to make sure there is no ambiguity because of default values.